Problem2.java

```diff
@@ -0,0 +1,43 @@
+/* Given a binary array, find the maximum length of a contiguous subarray with equal number of 0 and 1. */
+
+// Time Complexity : O(n) using hashmap
+// Space Complexity : O(1) in place
+// Did this code successfully run on Leetcode : Yes
+// Any problem you faced while coding this : No
```

Comment:
Space complexity will not be constant as you are using hashmap also

Reply:
Thanks for your observation. I have updated space complexity.

Reply:
good